Ghidra/Features/Decompiler/src/main/java/ghidra/app/plugin/core/decompile/actions/ShiftPointerAction.java

```diff
@@ -0,0 +1,208 @@
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ghidra.app.plugin.core.decompile.actions;
+
+import java.awt.event.KeyEvent;
+
+import docking.action.KeyBindingData;
+import docking.action.MenuData;
+import docking.widgets.OptionDialog;
+import docking.widgets.dialogs.NumberInputDialog;
+import ghidra.app.decompiler.*;
+import ghidra.app.decompiler.component.DecompilerUtils;
+import ghidra.app.plugin.core.decompile.DecompilerActionContext;
+import ghidra.framework.plugintool.PluginTool;
+import ghidra.program.model.data.DataType;
+import ghidra.program.model.data.DataTypeManager;
+import ghidra.program.model.data.Pointer;
+import ghidra.program.model.data.PointerDataType;
+import ghidra.program.model.listing.Function;
+import ghidra.program.model.listing.Program;
+import ghidra.program.model.pcode.*;
+import ghidra.program.model.symbol.SourceType;
+import ghidra.util.Msg;
+import ghidra.util.UndefinedFunction;
+import ghidra.util.exception.*;
+
+/**
+ * Action to shift a pointer by a specific offset
+ */
+public class ShiftPointerAction extends AbstractDecompilerAction {
+
+	public ShiftPointerAction() {
+		super("Shift Pointer");
+		setDescription("Shift selected pointer by an offset");
+		setPopupMenuData(new MenuData(new String[] { "Shift Pointer" }, "Decompile"));
+		setKeyBindingData(new KeyBindingData(KeyEvent.VK_T, 0));
+	}
+
+	/**
+	 * Retype a global or local symbol
+	 * @param program the program
+	 * @param highSymbol the symbol to retype
+	 * @param exactSpot the exact spot to retype
+	 * @param dt the new data type
+	 * @param tool the PluginTool
+	 */
+	private void retypeSymbol(Program program, HighSymbol highSymbol, Varnode exactSpot,
+			DataType dt, PluginTool tool) {
+		HighFunction hfunction = highSymbol.getHighFunction();
+
+		boolean commitRequired = checkFullCommit(highSymbol, hfunction);
+		if (commitRequired) {
+			exactSpot = null;		// Don't try to split out if commit is required
+		}
+
+		if (exactSpot != null) { // The user pointed at a particular usage, not just the vardecl
+			try {
+				HighVariable var = hfunction.splitOutMergeGroup(exactSpot.getHigh(), exactSpot);
+				highSymbol = var.getSymbol();
+			}
+			catch (PcodeException e) {
+				Msg.showError(this, tool.getToolFrame(), "Shift Failed", e.getMessage());
+				return;
+			}
+		}
+		DataTypeManager dataTypeManager = program.getDataTypeManager();
+		boolean successfulMod = false;
+		int transaction = program.startTransaction("Shift Pointer");
+		try {
+			if (dt.getDataTypeManager() != dataTypeManager) {
```

Comment:
Is reference equality correct here?

Reply:
I just copied the code from RetypeLocalAction, and it also worked with globals, so I assume it has a purpose